extensions/Maps/includes/Maps_Geocoder.php

```diff
@@ -111,7 +111,21 @@ public function hasAlias( $alias ) {
 	 * @return array or false
 	 */
 	public function geocode( $address ) {
-		$response = ExternalHttp::get( $this->getRequestUrl( $address ) ); # Wikia change
+
+		// MAIN-6654: cache requests to this API because it has daily limits
+		global $wgMemc;
+		$key = wfMemcKey( "geocoder", $address );
+		$response = $wgMemc->get( $key );
+		if ( empty($response) ) {
+			$response = ExternalHttp::get( $this->getRequestUrl( $address ) ); # Wikia change
+			// MAIN-6654: make sure we do not return or store a bad response because it causes errors later
+			if ( substr( $response, "OVER_QUERY_LIMIT" ) !== false ) {
```

Comment:
Can we check exactly for status instead using substr?
probability for occurrence this string in correct response is low, but let's check it in proper way.
Also is there a different status for OVER_QUERY_LIMIT? `429 Too Many Requests` maybe